src/core/crypto.rs

```diff
@@ -61,6 +61,26 @@ fn rsa_public_key(
     }
 }
 
+fn ec_public_key(
+    key: &CoreJsonWebKey,
+) -> Result<(&Base64UrlEncodedBytes, &Base64UrlEncodedBytes, &CoreJsonCurveType), String> {
+    if *key.key_type() != CoreJsonWebKeyType::EllipticCurve {
+        Err("EC key required".to_string())
+    } else if let Some(x) = key.x.as_ref() {
+        if let Some(y) = key.y.as_ref() {
+            if let Some(crv) = key.crv.as_ref() {
+                Ok((x,y, crv))
+            } else {
+                Err("CurveType is missing".to_string())
+            }
+        } else {
+            Err("EC `x` part is missing".to_string())
```

Comment:
I think this error should say `y`. the code above is structured similarly, but we'd probably avoid this sort of bug by expressing it as:
```
let x = key.x.as_ref().ok_or_else(|| Err("EC `x` part is missing".to_string()))?;
let y = key.y.as_ref().ok_or_else(|| Err("EC `y` part is missing".to_string()))?;
let crv = key.crv.as_ref().ok_or_else(|| Err("EC `crv` part is missing".to_string()))?;
Ok((x, y, crv))
```